helm/oauth2-proxy/Chart.yaml

```diff
@@ -1,5 +1,5 @@
 name: oauth2-proxy
-version: 3.3.2
+version: 3.3.3
```

Comment:
I think this is a breaking change, we should advertise it (major version bump + requiring kubernetes 1.xx in chart.yaml + readme + explain how to upgrade in the README).

People with existing Releases with `ingress.extraPaths` values will get impacted and need to adapt their values.

Reply:
I totally agree

Reply:
Can check and fail if ingress.extraPaths has old format..
See example from jenkins chart https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/templates/deprecation.yaml

Reply:
I like the deprecation.yaml approach :+1:
I will propose one in this MR and do a version bump to v4.0.0